packages/stylable.io/docs/guides/migration-v3.md

```diff
@@ -0,0 +1,138 @@
+---
+id: guides/migration-v3
+title: Migrating to Stylable v3
+layout: docs
+---
+
+This guide is intended to help migrate stylable version 1 or 2 to stylable version 3.
+It is mainly geared towards stylable integration within React.
+
+# Suggested steps of migration
+
+Follow these steps for a smooth transition. Each step is explained in
+more detail below.
+
+1. Update dependencies
+1. Update `.st.css` file imports
+1. Update usage in React components
+1. Update tests
+
+## Update dependencies
+
+Ensure you have v3 stylable dependencies available. You may need to
+update your `package.json` or ensure that other dependencies bring you
+v3 stylable:
+
+- `@stylable/cli`
+- `@stylable/core`
+- `@stylable/runtime`
+- `@stylable/node`
+- `@stylable/webpack-plugin`
+
+
+> Note: all stylable packages in v3 are scoped under `@stylable` namespace. 
+> if you have dependency like `stylable` (without namespace),
+> it is a different one and should be changed to namespaced version.
+
+## Update `.st.css` file imports
+
+Prior to v3 all `.st.css` files would export `style` function. In v3
+this has changed: `.st.css` files now export an object.
+
+List of all exported keys:
+
+```
+import {
+  classes,
+  cssStates,
+  keyframes,
+  namespace,
+  st,
+  stVars,
+  style,
+  vars,
+} from './style.st.css';
+```
+
+This means that all imports of `.st.css` files have to be changed, for example:
+
+```diff
+-import style from './Component.st.css';
++import { style, classes, /* ... */ } from './Component.st.css';
+```
+
+However, most often `{ style, classes }` is enough:
+
+`import { style, classes } from './Component.st.css';`
+
+## Update usage in React components
+
+Once `.st.css` imports are updated, React components should be updated too:
+
+```diff
+-<div {...style('root', states, props)} /> />
++<div className={style(classes.root, states, props.className)} />
+```
+
+There is a subtle but very important nuance in this change.
+
+Stylable v1 and v2 used spread pattern. It would take the output of
```

Comment:
only v1 uses the spread

Reply:
hey, thanks for reviewing! However, i believe you're looking at older PR version, @tomrav has already helped me with most of the mistakes